Sources/Private/Experimental/Animations/CAAnimation+TimingConfiguration.swift

```diff
@@ -68,6 +68,24 @@ extension CAAnimation {
     clippingParent.fillMode = .both
     clippingParent.isRemovedOnCompletion = false
 
+    // We can pause the animation on a specific frame by setting the root layer's
+    // `speed` to 0, and then setting the `timeOffset` for the given frame.
+    //  - For that setup to work properly, we have to set the `beginTime`
+    //    of this animation to a time slightly before the current time.
+    //  - It's not really clear why this is necessary, but `timeOffset`
+    //    is not applied correctly without this configuration.
+    let currentTime = layer.convertTime(CACurrentMediaTime(), from: nil)
+    clippingParent.beginTime = currentTime - .leastNonzeroMagnitude
```

Comment:
This is somewhat cursed, but it fixes an issue where the `timeOffset` would be applied incorrectly. Without the `- .leastNonzeroMagnitude`, the actual displayed frame is somehow dependent on the frame of the animation when switching from noninteractive to interactive playback:

| Before | After |
| ---- | ---- |
| ![ezgif-2-1fde89d872](https://user-images.githubusercontent.com/1811727/149197905-15021cd2-bc3a-44a4-9586-dd1067136714.gif) | ![2022-01-12 10 10 04](https://user-images.githubusercontent.com/1811727/149197615-0984ed96-52d4-4a77-a857-e24958f0c3ea.gif) |

Reply:
The fact that `clippingParent.beginTime = currentTime - .leastNonzeroMagnitude` works but `clippingParent.beginTime = currentTime` doesn't kinda blows my mind, considering `Double.leastNonzeroMagnitude` is `5e-324`.

Reply:
Huh how strange, glad you found a path forward 😅